docs/getting-started/provider-aws.md

```diff
@@ -0,0 +1,219 @@
+---
+title: AWS Quickstart
```

Comment:
I know this is a bit of a controversial opinion, but I question whether https://crossplane.io/docs is the right place for guides that are focused on a particular provider. I wonder if we could get folks up-to-speed with what Crossplane is all about faster by focusing on a simpler scenario that demonstrated what Crossplane was all about.

I like how the Terraform docs approach this. They have getting started guides for the major cloud providers, but before you get into provider specific details they take you through a little "deploy a Docker container" scenario - e.g.:

* https://developer.hashicorp.com/terraform/tutorials/aws-get-started starts with "What is IaC with Terraform?" which takes me to an interactive tutorial to spin up a Docker container.
* (Oddly) the second entry in https://developer.hashicorp.com/terraform/tutorials/aws-get-started is https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli, which again shows me how to use Terraform to spin up a Docker container.

By following these tutorials I've run some Terraform commands, seen what the syntax looks like, and got a quick win (actually deployed something) without having to even have access to a cloud account, figure out credentials, etc.

If we adopted this approach it would potentially allow us to keep https://crossplane.io/docs "provider agnostic", with provider-specific quick start guides living closer to the provider implementations (e.g. in their Git repos and/or Marketplace entries). I feel this would encourage a broader set of quick start guides (not just the "big three" cloud providers and increase the chances of provider-specific docs stay up-to-date with their providers.

To be a little more concrete, what I imagine is a getting started flow something like:

1. Install Crossplane.
2. Create an MR that orchestrates some simple, credential-free external system.
3. Show that Crossplane actually changes the external system, corrects drift, etc.
4. Create a Configuration that builds an abstraction (claim, Composition, etc) around the MR from step 2.
5. Create a claim.

Obviously the gap here would be finding/building such a simple, credential-free external system. We can't deploy a Docker container on your laptop like Terraform does, and something like https://github.com/crossplane-contrib/provider-nop doesn't illustrate how Crossplane actually changes an external system.

Reply:
Agreed. Discussions with @hasheddan and @bassam the conclusion was "We need _something_ to show how Crossplane works". In this case the _something_ is a provider. 

We are using the big-three cloud providers to get a large set of users. If other provider maintainers want to contribute quickstart guides in the future we should welcome this. If we have dozens of guides then it's a good problem to have and we can solve it then. 

Regarding your "flow" this is exactly what I envision. This may be in a single large quickstart or across two. Right now my preference is across two guides so we don't have a 40-page "quick" start.

If you have additional questions about using providers in the docs it's worth a meeting between you, me and @bassam  to discuss.